setup.py

```diff
@@ -29,10 +29,31 @@
                  install_requires=[
                      'numpy',
                      'pybind11>=2.5.0',
-                     'colorama',
                      'setuptools',
                      'astor',
                  ],
+                 tests_requires=[  # To run test with Taichi:
+                     'pytest',
+                     'pytest-rerunfailures',
+                     'pytest-xdist',
+                     'autograd',
+                 ],
+                 extra_requires={
+                     'dev': {  # Required for Taichi developers:
+                         'GitPython',
+                         'coverage',
+                         'sphinx',
+                     },
+                     'torch': {  # Optional PyTorch support:
+                         'torch',
+                     },
+                     'shell': {  # To use Taichi in interactive shell:
+                         'dill',
+                     },
+                     'color': {  # To make Taichi output colorful:
+                         'colorama',
+                     },
+                 },
```

Comment:
Is this what we want for having `taichi[color]`?

Reply:
Yes, I believe above is what we need for it, you could test it easily locally too with `pip install -e .[color]` in an isolated environment.

One nit: the above setup provides perhaps too many choices/combinations, and may cause confusion to the users, I'd suggest we only make distinguish between `dev` and `bundle` (and maybe `testing` as well), e.g. `pip install taichi[dev]` v.s. `pip install taichi[bundle]`

Reply:
> One nit: the above setup provides perhaps too many choices/combinations, and may cause confusion to the users, I'd suggest we only make distinguish between `dev` and `bundle` (and maybe `testing` as well), e.g. `pip install taichi[dev]` v.s. `pip install taichi[bundle]`

I agree with this... I think having 2-3 options is good enough. Otherwise it's too complex to install...